Map/MapBase.gd

```diff
@@ -1,7 +1,10 @@
 extends Resource
-class_name Map
-## Class for a map. Provides classes for the map
+class_name MapBase
+## Class for the map. Provides functions for the map
 ##
 ## Holds basic functionality for a map
-@export var floors: Array[Floor]
+
+@export var rooms: Array = [] ## 2D array of rooms on the map. this will be assigned when generating rooms
+@export var width:int = 3
```

Comment:
this supposes that the width of the map is constant. This is a supposition we can make on height, but as seen in the drawing of the issue #6 the width can change.
The width should be an array instead of a singular value. There is also a condition on the array which is that each floor can only be 2 more wide than the floor before it (to be sure we can always access all the rooms on the next floor).
This wasn't explicitly discussed in the issue, i will add a comment now to talk about this with more details.